lib/contracts/builtin_contracts.rb

```diff
@@ -405,6 +405,7 @@ def initialize(contract_hash)
 
       def valid?(arg)
         return false unless arg.is_a?(Hash)
+        return false unless arg.keys.sort.eql?(contract_hash.keys.sort)
```

Comment:
`sort` is missing for ruby 1.8.7, but is there any point of supporting this version? Also what do you think about comparing arrays sizes instead?